tests/compile-fail/large_enum_variant.rs

```diff
@@ -0,0 +1,39 @@
+#![feature(plugin)]
+#![plugin(clippy)]
+
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#![deny(large_enum_variant)]
+
+enum LargeEnum {
+    A(i32),
+    B([i32; 8000]), //~ ERROR large enum variant found on variant `B`
+}
+
+enum GenericEnum<T> {
+    A(i32),
+    B([i32; 8000]), //~ ERROR large enum variant found on variant `B`
+    C([T; 8000]),
+    D(T, [i32; 8000]), //~ ERROR large enum variant found on variant `D`
+}
+
+trait SomeTrait {
+    type Item;
+}
+
+enum LargeEnumGeneric<A: SomeTrait> {
+    Var(A::Item), // regression test, this used to ICE
```

Comment:
I got tired of fixing multiple commits during rebasing and simply merged them into one